From c511f5dcd7d61d33b76429a902a2be628bfaf8ae Mon Sep 17 00:00:00 2001
From: yzhu1 <yanjun.zhu@windriver.com>
Date: Wed, 22 Jul 2015 15:14:19 +0800
Subject: [PATCH] dpa: set skb truesize to the skb length if jumbo is enabled

This patch is from FSL SDK 1.6.

After the jubmo is set to 9600, the truesize of skb should be set
to the skb length before these skb packets are sent to netif_receive_skb.

Signed-off-by: yzhu1 <yanjun.zhu@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
index a47dfab..3c51848 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
@@ -265,6 +265,8 @@ struct sk_buff *_dpa_cleanup_tx_fd(const struct dpa_priv_s *priv,
 	return skb;
 }
 
+#define JUMBO_LENGTH	9600
+
 /*
  * Build a linear skb around the received buffer.
  * We are guaranteed there is enough room at the end of the data buffer to
@@ -292,6 +294,9 @@ static struct sk_buff *__hot contig_fd_to_skb(const struct dpa_priv_s *priv,
 	if (unlikely(!skb))
 		return NULL;
 
+	if (JUMBO_LENGTH == dpa_get_max_frm())
+		skb->truesize = SKB_TRUESIZE(dpa_fd_length(fd));
+
 	/* Make sure forwarded skbs will have enough space on Tx,
 	 * if extra headers are added.
 	 */
-- 
1.7.5.4

